docs/utilization.md

```diff
@@ -2,157 +2,201 @@
+out_of_office = 0
 billing_expectation = 0.80
 
-total_hours = billable_hours + non_billable_hours
-# 24.0 + 16.0 = 40
+target_billable_hours = (hours_in_week - out_of_office) * billing_expectation
+# (40 - 0) * 0.80 = 32
 
-target_billable_hours = total_hours * billing_expectation
-# 40 * 0.80 = 32
+billable_allocation_hours = billable_allocation * target_billable_hours
+# .25 * 32 = 8
 
-utilization = billable_hours / target_billable_hours
-# 24 / 32 = 0.75
+utilization = (billable_hours + billable_allocation_hours) / target_billable_hours
+# (24 + 8) / 32 = 1
 ```
 
 ### Week 2
-Stacey's `utilization` for Week 2 is 109%.
-Stacey tocked 28 hours when their target was ~25 billable hours.
+Stacey's `utilization` for Week 2 is 117%.
+
+Stacey tocked 30 billable hours when their target was ~25 billable hours.
+
+Stacey's target was less than the previous week because they were `Out of Office` for 8 hours.
+
+```python
+# Week 2
+billable_hours = 30.0
+non_billable_hours = 2.0
+billable_allocation = 0
+out_of_office = 8
+billing_expectation = 0.80
+
+target_billable_hours = (hours_in_week - out_of_office) * billing_expectation
+# (40 - 8) * 0.80 = 25.6
 
-Stacey's target was less than the previous week because they were `Out of Office` for 8 hours in week 2.
+billable_allocation_hours = billable_allocation * target_billable_hours
+# 0 * 25.6 = 0
+
+utilization = (billable_hours + billable_allocation_hours) / target_billable_hours
+# (30 + 0) / 25.6 = 1.17
+```
+
+### Week 3
+Stacey's `utilization` for Week 3 is 100%.
+
+Stacey tocked 100% to a weekly billing project and no hours to hourly billable projects.
 
 ```python
 # Week 2
-billable_hours = 28.0
-non_billable_hours = 4.0
+billable_hours = 0
+non_billable_hours = 0
+billable_allocation = 1.0
+out_of_office = 0
 billing_expectation = 0.80
 
-total_hours = billable_hours + non_billable_hours
-# 28.0 + 4.0 = 32
+target_billable_hours = (hours_in_week - out_of_office) * billing_expectation
+# (40 - 8) * 0.80 = 32
 
-target_billable_hours = total_hours * billing_expectation
-# 40 * 0.80 = 25.6
+billable_allocation_hours = billable_allocation * target_billable_hours
+# 1.0 * 32 = 32
 
-utilization = billable_hours / target_billable_hours
-# 28 / 25.6 = 1.09
+utilization = (billable_hours + billable_allocation_hours) / target_billable_hours
+# (0 + 32) / 32 = 1
 ```
 
 ### Weeks 1-4
-Stacey's `utilization` over the last 4 weeks is 95%. Stacey tocked 116 of an expected ~122 billable hours over the 4 week period.
+Stacey's `utilization` over the last 4 weeks is ~104%.
 
-Stacey had eight hours of `Out of Office` time in week 2, which is excluded from all totals. Notice the `total_hours` is 152 for the 4 weeks.
+Notice how values for each week are calculated independently and then summed to produce a utilization number over the 4-week time frame. Because the target billable hours denominator may fluctuate week-to-week, we can not simply take the average of each week's utilization percentages.
```

Comment:
I think that we can just take the average!  I worked this out on paper a few weeks ago to make sure that I wasn't misremembering how math works!  I think maybe what we should mention is that we can't (???) because if someone with no billable expectation did bill hours, their utilization of 0% would not roll up as accurately as if we were summing in those hours with weeks when they had a billable expectation?  Does that make sense?

Reply:
@jduss4 and I discussed offline. We decided that an average of utilization rates will not be accurate if we are calculating utilization as number of billable hours divided by number of target hours across multiple timecards. See the example below.

An average _would_ be accurate if we were trying to answer a question more like "How has a user been allocated to a project over time?" But that's not the case here. It's been decided that utilization rate should be calculated as explained above.

Example:

Value | Week 1 | Week 2 | Aggregate | Average
------|--------|---------|-------|---
Hours worked | 40 | 20 | 60 | 30
Billable hours | 32 | 8 | 40 | 20
Target hours | 32 | 16 | 48 | 24
Utilization | 100% | 50% | 83% | 75%